spacy/language.py

```diff
@@ -2084,10 +2075,12 @@ def _resolve_component_status(
         if enable:
             if isinstance(enable, str):
                 enable = [enable]
-            to_disable = [
-                pipe_name for pipe_name in pipe_names if pipe_name not in enable
-            ]
-            if disable and disable != to_disable:
+            to_disable = {
+                *[pipe_name for pipe_name in pipe_names if pipe_name not in enable],
+                *disable,
+            }
+            # If any pipe to be enabled is in to_disable, the specification is inconsistent.
+            if any([pipe_to_enable in to_disable for pipe_to_enable in enable]):
```

Comment:
I was surprised that this check didn't use a set operation?

Reply:
No particular reason for this. Replaced with a set overlap check.

Reply:
I was just surprised because I initially thought something more complicated must be going on, but it wasn't, right?

Reply:
Before this PR we [checked for equivalence](https://github.com/explosion/spaCy/pull/11745/files#diff-fe971afb0a4f4d99848a30cfc2708c359e30703084ea5220629928ac9015f15dL2090) between the sets of pipes defined via `exclude` and those via `include` (and raised an error in case of inequality).

With our current assumptions, I think a conflict should occur if and only if the sets of pipes to be enabled (in `enable(d)`) and pipes to be disabled (pipes not in `enable(d)` + pipes in `disable(d)`). Do you agree?

Reply:
The current check is what I expected.